src/components/settings/SpendingLimits/NewSpendingLimit/steps/SpendingLimitForm.tsx

```diff
@@ -110,11 +113,26 @@ export const SpendingLimitForm = ({ data, onSubmit }: Props) => {
             <NumberField
               label={errors.amount?.message || 'Amount'}
               error={!!errors.amount}
+              InputProps={{
+                endAdornment: (
+                  <InputValueHelper onClick={onMaxAmountClick} disabled={!selectedToken}>
+                    Max
+                  </InputValueHelper>
+                ),
+              }}
+              // @see https://github.com/react-hook-form/react-hook-form/issues/220
+              InputLabelProps={{
+                shrink: !!watch('amount'),
+              }}
               required
               {...register('amount', {
                 required: true,
-                validate: (val) =>
-                  validateAmount(val) || _validateSpendingLimit(val, selectedToken?.tokenInfo.decimals),
+                validate: (val) => {
+                  const decimals = selectedToken?.tokenInfo.decimals
+                  return (
+                    validateLimitedAmount(val, decimals, totalAmount.toString()) || validateDecimalLength(val, decimals)
```

Comment:
It should also be possible to set spending limits higher than that of the total owned.